optimizely/config_manager.py

```diff
@@ -269,9 +278,14 @@ def is_running(self):
 
     def _run(self):
         """ Triggered as part of the thread which fetches the datafile and sleeps until next update interval. """
-        while self.is_running:
-            self.fetch_datafile()
-            time.sleep(self.update_interval)
+        try:
+          while self.is_running:
+              self.fetch_datafile()
+              time.sleep(self.update_interval)
+        except (OSError, OverflowError) as err:
+            self.logger.error('Error in time.sleep. '
+                              'Provided update_interval value may be too big. Error: {}'.format(str(err)))
+            raise
```

Comment:
Should we still be raising in this case or just pass it to the error handler? Not sure if raising from another thread might be an issue

Reply:
I think if we hit either of these errors we should just raise so that the user sees this and fixes it. We cannot leave it silent.

Reply:
Won't this crash the application though? The job of the error handler in this case would be raise visibility (if an error handler is provided), otherwise it'll just show up in the logs (again, if a logger is provided)

Reply:
If the value (which is provided the first time) is too high to cause `OverflowError` or `OSError` the SDK won't be initialized in the first place.

Reply:
I see. Wouldn't that make this check redundant then?

Reply:
Yeah, it may seem that way, but unfortunately `time.sleep` has different limits. See here: https://github.com/python/cpython/blob/2.7/Modules/timemodule.c#L1030-L1155

We are not imposing any limit at init after our discussion last week. I was in favor on imposing a 1 day limit, but may be it is ok for `time.sleep` to fail to let people know that they should change it to a reasonable value. That way they can adjust for their platform accordingly.

Reply:
Got it. Thanks for the explanation.

Reply:
👍